physical/tvf.go

```diff
@@ -221,6 +245,15 @@ func (node *TableValuedFunction) Metadata() *metadata.NodeMetadata {
 			cardinality = metadata.BoundedFitsInLocalStorage
 		}
 		return metadata.NewNodeMetadata(cardinality, octosql.NewVariableName("window_end"))
+	case "max_diff_watermark":
+		var cardinality metadata.Cardinality
+		source, err := node.getArgumentTable(octosql.NewVariableName("source"))
+		if err == nil {
+			cardinality = source.Metadata().Cardinality()
+		} else {
+			cardinality = metadata.BoundedFitsInLocalStorage
+		}
+		return metadata.NewNodeMetadata(cardinality, octosql.NewVariableName("time_field"))
```

Comment:
That's not true, we don't set a "time_field" variable. Just pass what you get from underneath.